src/strands/hooks/events.py

```diff
@@ -97,14 +97,17 @@ class BeforeToolCallEvent(HookEvent):
             to change which tool gets executed. This may be None if tool lookup failed.
         tool_use: The tool parameters that will be passed to selected_tool.
         invocation_state: Keyword arguments that will be passed to the tool.
+        cancel: A user defined message that when set, will cancel the tool call.
+            The message will be placed into a tool result with an error status.
     """
 
     selected_tool: Optional[AgentTool]
     tool_use: ToolUse
     invocation_state: dict[str, Any]
+    cancel: Optional[str] = None
```

Comment:
nit: cancel_tool?
```suggestion
    cancel_tool: str | None = None
```

also wondering if a flag or a function makes more sense here? Curious on your thoughts

Reply:
The flag makes sense, but I still think a bool + string would be nice. Just a simple sensible default so I dont need to think about what a good cancel message for a tool is